internal/api/mfa.go

```diff
@@ -89,25 +89,33 @@ func (a *API) EnrollFactor(w http.ResponseWriter, r *http.Request) error {
 		issuer = params.Issuer
 	}
 
-	// Read from DB for certainty
 	factors, err := models.FindFactorsByUser(a.db, user)
 	if err != nil {
 		return internalServerError("error validating number of factors in system").WithInternalError(err)
 	}
-
-	if len(factors) >= int(config.MFA.MaxEnrolledFactors) {
-		return forbiddenError("Enrolled factors exceed allowed limit, unenroll to continue")
-	}
-
+	factorCount := len(factors)
 	numVerifiedFactors := 0
+
+	// Cleanup inactive  factors
 	for _, factor := range factors {
+		if factor.IsExpired(config.MFA.FactorExpiryDuration) {
+			if err := a.db.Destroy(factor); err != nil {
+				return internalServerError("error deleting factors").WithInternalError(err)
+			}
+			// We adjust length of factors as destroying it in the DB doesn't remove it from the array
+			factorCount -= 1
+		}
```

Comment:
Why don't we just do a SQL query in the database that deletes them? You can also use that query in the cleanups that run on other tables too. Iteratively destroying it can take a _very_ long time and suck all performance out of the system.

Reply:
Sound good, let's do that